src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1069,14 +1069,22 @@ void primitive_inst::save(cldnn::BinaryOutputBuffer& ob) const {
     ob << can_be_optimized();
     ob << can_share_buffer();
     ob << is_constant();
+    auto users = get_node().get_users();
+    bool is_output_event = is_any_user_cpu(users) || get_node().is_output();
+    ob << is_output_event;
 
     if (type() == cldnn::data::type_id()) {
         return;
     }
 
-    ob << _outputs[0]->get_layout();
-    const auto _allocation_type = _outputs[0]->get_allocation_type();
-    ob << make_data(&_allocation_type, sizeof(_allocation_type));
+    if (_outputs[0] == nullptr) {
+        ob << true;
```

Comment:
How is this different from mem_allocated ?

Reply:
`mem_allocated` does not tell if an operation has an output memory.
For example, in the case of implicit concat, `mem_allocated` is false and output is not null.
So, we can query output layout for the concat operation.
But in the case of the read_value operation, `mem_allocated` is false and output is also null.
Output for the read_value operation is allocated by any external logic.

Reply:
Thanks for the explanation. Then for that purpose, can we just infer as 
( !mem_allocated && can_be_optimized) => has_output mem
( !mem_allocated && !can_be_optimized) => no output mem ?

I'd like to have # of similar variables as less as possible, becasue it would be very confusing in the future :D

Reply:
In the case of `mutable_data`, mem_allocated is false, can_be_optimized is false, and output is not null.
I agree with your opinion, but I'm not sure if it's possible to replace it with a combination of multiple variables.

Reply:
I added comments for the difference between mem_allocated and is_output_null to avoid future confusion. Thank you.